webapp/sources/rudder/rudder-core/src/main/scala/com/normation/rudder/services/servers/NewNodeManager.scala

```diff
@@ -460,108 +461,12 @@ trait ComposedNewNodeManager extends NewNodeManager with NewNodeManagerHooks {
-        rollback(unitAcceptors, Seq(sm), modId, actor)
-        e
-    }
-  }
 
   override def accept(id: NodeId, modId: ModificationId, actor:EventActor) : Box[FullInventory] = {
-    //
-    // start by retrieving all sms
-    //
-    val sm = smRepo.get(id, PendingInventory).toBox match {
-      case Full(Some(x)) => x
-      case Full(None) =>
-        return Failure(s"Can not accept not found inventory with id: '${id.value}'")
-      case eb: EmptyBox =>
-        return eb ?~! "Can not accept not found inventory with id %s".format(id)
-      }
-
-    //
-    //execute pre-accept phase for all unit acceptor
-    // stop here in case of any error
-    //
-    unitAcceptors.foreach { unitAcceptor =>
-      unitAcceptor.preAccept(Seq(sm), modId, actor) match {
-        case Full(seq) => //ok, cool
-          NodeLogger.PendingNode.debug("Pre accepted phase: %s".format(unitAcceptor.name))
-        case eb:EmptyBox => //on an error here, stop
-          val id = sm.node.main.id.value
-          val msg = s"Error when trying to add node with id '${id}'"
-          val e = eb ?~! msg
-          NodeLogger.PendingNode.error(e.messageChain)
-          NodeLogger.PendingNode.debug(s"Node with id '${id}' was refused during 'pre-accepting' step of phase '${unitAcceptor.name}'")
-          //stop now
-          return e
-      }
-    }
-
-    //
-    //now, execute unit acceptor
-    //
-
-    //build the map of results
-    val acceptationResults = acceptOne(sm, modId, actor) ?~! "Error when trying to accept node %s".format(sm.node.main.id.value)
-
-    //log
-    acceptationResults match {
-      case Full(sm) => NodeLogger.PendingNode.debug("Unit acceptors ok for %s".format(id))
-      case eb:EmptyBox =>
-        val e = eb ?~! "Unit acceptor error for node %s".format(id)
-        NodeLogger.PendingNode.error(e.messageChain)
-        return eb
-    }
-
-    //
-    //now, execute global post process
-    //
-    (sequence(unitAcceptors) { unit =>
-      try {
-        unit.postAccept(Seq(sm), modId, actor)
-      } catch {
-        case e:Exception => Failure(e.getMessage, Full(e), Empty)
-      }
-    }) match {
-      case Full(seq) => //ok, cool
-        // Update hooks for the node
-        afterNodeAcceptedAsync(id)
-
-        NodeLogger.PendingNode.info(s"New node accepted and managed by Rudder: ${id.value}")
-        cacheToClear.foreach { _.clearCache }
-
-        // Update hooks for the node - need to be done after cache cleaning
-        afterNodeAcceptedAsync(id)
-
-        updateDynamicGroups.startManualUpdate
-        acceptationResults
-      case e:EmptyBox => //on an error here, rollback all accpeted
-        NodeLogger.PendingNode.error((e ?~! "Error when trying to execute accepting new server post-processing. Rollback.").messageChain)
-        rollback(unitAcceptors, Seq(sm), modId, actor)
-        //only update results that where not already in error
-        e
+    accept(List(id), modId, actor, "rudder-ui").flatMap {
```

Comment:
shouldn't we have the actor ip ? (if available?)

Reply:
we don't have any mean to get it for now. It would be a nice little enhancement for a future version